cogito/put_test.go

```diff
@@ -14,47 +15,104 @@ import (
 	"gotest.tools/v3/assert"
 )
 
+var (
+	baseSource = cogito.Source{
+		Owner:       "the-owner",
+		Repo:        "the-repo",
+		AccessToken: "the-token",
+	}
+
+	baseParams = cogito.PutParams{State: cogito.StatePending}
+
+	basePutInput = cogito.PutInput{
+		Source: baseSource,
+		Params: baseParams,
+	}
+)
+
+type MockPutter struct {
+	loadConfigurationErr error
+	processInputDirErr   error
+	outputErr            error
+	sinkers              []cogito.Sinker
```

Comment:
This allows to control the mock at init time and specify if and when to return an error.